src/test/java/org/aksw/iguana/cc/query/list/QueryListTest.java

```diff
@@ -100,9 +101,13 @@ public static List<Arguments> data() throws IOException {
         return out;
     }
 
+    @Test
     public void testIllegalArguments() {
         assertThrows(NullPointerException.class, () -> new InMemQueryList(null));
-        assertThrows(NullPointerException.class, () -> new FileBasedQueryList(null));
+        assertThrows(NullPointerException.class, () -> {
+            final var ql = new FileBasedQueryList(null);
+            ql.size();
```

Comment:
what is the ql.size() call good for?

Reply:
The constructor of FileBasedQueryList doesn't throw NullPointerException on creation. It only does if you call an additional method. I just added that in quickly to make the test work, but I think explicitly checking for empty parameters in the constructors would be the better thing to do.